src/main.rs

```diff
@@ -196,6 +195,15 @@ fn main() -> Result<()> {
                 Resources::app => apps::create(&context, id, data, file),
                 Resources::device => {
                     let app_id = arguments::get_app_id(&command.unwrap(), &context)?;
+
+                    if command.unwrap().is_present(&Other_flags::cert) {
+                        id = format!("CN={}, O=Drogue IoT, OU={}", id, app_id);
```

Comment:
you can use variable shadowing instead of making the variable `mut` :
 ```suggestion
                       let id = format!("CN={}, O=Drogue IoT, OU={}", id, app_id);
```